llvm/tools/llvm-exegesis/lib/Target.cpp

```diff
@@ -35,6 +35,19 @@ const ExegesisTarget *ExegesisTarget::lookup(Triple TT) {
   return nullptr;
 }
 
+const char *
+ExegesisTarget::getIgnoredOpcodeReasonOrNull(const LLVMState &State,
+                                             unsigned Opcode) const {
+  const MCInstrDesc &InstrDesc = State.getIC().getInstr(Opcode).Description;
+  if (InstrDesc.isPseudo() || InstrDesc.usesCustomInsertionHook())
+    return "Unsupported opcode: isPseudo/usesCustomInserter";
+  if (InstrDesc.isBranch() || InstrDesc.isIndirectBranch())
```

Comment:
Not related to this patch, but RISCV::JAL and RISCV::JALR can be call or branch depending on which registers it uses. RISCV::JALR can also be a return. Those instruction don't have any of these flags set because of this.

What's causing those to be ignored today?

Reply:
> What's causing those to be ignored today?

I don't think JAL and JALR are explicitly ignored. If anything, the snippets for these two opcodes will just crash silently (with a message shown in the result, of course) inside the sandbox built by JIT without affecting other snippets.